apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,521 @@
+				goto error_with_frt_status;
+			}
+
+			fd = open(g_file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			ret = write(fd, g_rw_buf, g_fsz);
+			if (ret != (g_fsz)) {
+				 printf("Unable to write to file: %s, ret: %d\n", g_file[0], ret);
+				 ret = ERROR;
+				 goto close_with_fd;
+			}
+			close(fd);
+
+			ret = unlink(g_file[0]);
+			if (ret != OK) {
+				printf("Failed to unlink file: %s, errno: %d\n", g_file[0], errno);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			ret = ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			if (ret < 0) {
+				goto error_with_frt_status;
+			}
+
+			itr_time = after.timeleft - before.timeleft;
+			printf("iteration %d, time: %lu us\n", i, itr_time);
+			tot_time += itr_time;
+		}
+		printf("\nFinal time: %lu us\n", tot_time);
+		break;
+
+	default:
+		break;
+	}
+
+	ret = ioctl(frt_fd, TCIOC_STOP, 0);
+	if (ret < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Timer : %d\n", errno);
+		ret = -1;
+		close(frt_fd);
+	}
+
+	close(frt_fd);
+	return ret;
+
+error_with_frt_status:
+	fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+	ret = -1;
+	goto close_with_frt;
+close_with_fd:
+	close(fd);
+close_with_frt:
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Free Run Timer: %d\n", errno);
+	}
+
+	close(frt_fd);
+
+	return ret;
+}
+
+static void show_usage() {
+	printf("\nusage: fs_performance [Timer device no.] [Test case] [File size(bytes)] [Buffer size(bytes)]\n");
+	printf("Timer Device no. :         FRT Timer device\n");
+	printf("Test Case        :         Test case to be executed\n");
+	printf("      1. File Creation Test : Creates a file of size \"File size\" in one go.\n");
+	printf("      2. Overwrite Test : Overwrite a file from start with buffer size \"Buffer size\".\n");
+	printf("      3. Append Test : Append 64 kb data to a file of size \"File size\".\n");
+	printf("      4. Read Test : Read a file of size \"File size\" with buffer size \"Buffer size\".\n");
+	printf("      5. Garbage Collection Effect Test : Create a file and unlink it to trigger GC.\n");
+	printf("File Size        :         File size(bytes) less than equal to 64kb\n");
+	printf("Buffer Size      :         Buffer size(bytes) less than equal to File size\n");
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int fs_performance_main(int argc, char *argv[])
+#endif
+{
+	int option = -1;
+	int ret = 0;
+	int timer_devno = 0;
+	int selected_opt;
+	g_fsz = MAX_FILE_SIZE;
+	g_bsz = MAX_RD_BUF_SIZE;
+
+	if (argc == 2 && !strncmp(argv[1], "--help", strlen("--help") + 1)) {
+		show_usage();
+		return OK;
+	}
+
+	if (argc > 4) {
+		timer_devno = atoi(argv[1]);
+		option = atoi(argv[2]);
+		if (option < 1 || option > 5) {
+			printf("Test case option should be within 1-5\n");
+			goto errout;
```

Comment:
1. The sentence of errout and usages are different.
2. I think no need to keep two different code for usages. Let's call show_usages() instead of errout.

Reply:
I agree,
In case of exiting example due to option discrepancy and not kernel task failure, lets show usage

Reply:
Changes made accordingly